Automattic/wp-calypso
# 2023 Pricing Grid: Use smallest unit for currencies in the plans pricing grid

Related to https://github.com/Automattic/wp-calypso/pull/81338#discussion_r1318929055

## Context

With the new storage add-ons feature, a user can add storage upgrades when they purchase a plan in onboarding. What this means is that there's a point where, behind the scenes, we need to calculate add-on prices + plan prices in a dynamic way. The arithmetic operation would ideally look something like $20 add-on + $20 plan = $40 total. There is, however, a hiccup.

Right now, `add-on` prices only return in the smallest unit ( Ex. USD $20 is represented in a value that corresponds to cents, so it's returned as the integer 2000 ). This is what's recommended by the payments team so as to avoid rounding errors from floating point calculations.

`plan` prices, however, are not handled in the smallest unit. ( Ex. USD $20 is represented as dollars, so it's handled as the integer 20 )

Fixing this representation mismatch is what we're trying to address in this PR. We are changing all plan prices to use smallest units for a given currency.

## Proposed Changes

* ~Ensure changes only target the `/start` flow~ from @southp: it's no longer the case :) 
* Changes all prices in the 2023 plan features grid to use the smallest unit a currency provides

## Screenshots
### Plans Features Grid
<img width="1477" alt="Screenshot 2023-09-12 at 5 56 06 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/f0fe60b0-2228-4b0b-9523-5d0c54eb1dec">

### Plans Comparison Grid
<img width="1326" alt="Screenshot 2023-09-12 at 5 56 14 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/ebf87221-9725-46eb-9c47-60c0fee7a327">

### Sticky CTA Buttons
<img width="1474" alt="Screenshot 2023-09-12 at 6 23 35 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/8db852ce-7747-43d9-8880-c938d086bcb2">

## Testing Instructions


* Check out this PR
* Navigate to `/start/plans`
* Verify that plans in the pricing headers and billing timeframes display as expected in the plans comparison grid and the plans features grid.
* Verify that prices display correctly in the sticky CTA buttons.
* Repeat verification with different WordPress currencies if possible USD, EUR, AUD, GBP, BRL, CAD, CHF, CZK, DKK, HKD, HUF, IDR, ILS, INR, JPY, MXN, NOK, NZD, PHP, PLN, RUB, SEK, SGD, THB, TWD, TRY
* Repeat verification with different term prices if available ( monthly vs. annually vs. 2yearly vs. 3yearly )
* Repeat verification in the following flows PCYsg-RDT-p2
* Repeat verification for the Calypso admin `/plans` page ( wordpress.com/plans/{SITE_SLUG} )

## Pre-merge Checklist
